test/Concurrency/Runtime/async_sequence.swift

```diff
@@ -0,0 +1,524 @@
+  expectEqual(result, [])
+}
+
+var AsyncCompactMapTests = TestSuite("AsyncCompactMap")
+
+AsyncCompactMapTests.test("non nil values") {
+  let result = await [1, 2, 3].async.compactMap { $0 }.collect()
+  expectEqual(result, [1, 2, 3])
+}
+
+AsyncCompactMapTests.test("nil values") {
+  let result = await [1, nil, 2, nil, 3, nil].async
+    .compactMap { $0 }.collect()
+  expectEqual(result, [1, 2, 3])
+}
+
+AsyncCompactMapTests.test("non nil values async") {
+  let result = await [1, 2, 3].async
+    .compactMap { await asynchronous($0) }.collect()
+  expectEqual(result, [1, 2, 3])
+}
+
+AsyncCompactMapTests.test("nil values async") {
+  let result = await [1, nil, 2, nil, 3, nil].async
+    .compactMap { await asynchronous($0) }.collect()
+  expectEqual(result, [1, 2, 3])
+}
+
+AsyncCompactMapTests.test("non nil values with throw") {
+  do {
+    let result = try await [1, 2, 3].async
+      .compactMap { try throwing($0) }.collect()
+    expectEqual(result, [1, 2, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncCompactMapTests.test("nil values with throw") {
+  do {
+    let result = try await [1, nil, 2, nil, 3, nil].async
+      .compactMap { try throwing($0) }.collect()
+    expectEqual(result, [1, 2, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncCompactMapTests.test("non nil values with async throw") {
+  do {
+    let result = try await [1, 2, 3].async
+      .compactMap { try await asynchronousThrowing($0) }.collect()
+    expectEqual(result, [1, 2, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncCompactMapTests.test("nil values with async throw") {
+  do {
+    let result = try await [1, nil, 2, nil, 3, nil].async
+      .compactMap { try await asynchronousThrowing($0) }.collect()
+    expectEqual(result, [1, 2, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncCompactMapTests.test("throwing") {
+  do {
+    _ = try await [1, nil, 2, nil, 3, nil].async
+      .compactMap { value throws -> Int? in
+        if value == 2 {
+          throw Failure()
+        }
+        return value
+      }
+      .collect()
+    expectUnreachable()
+  } catch {
+
+  }
+}
+
+var AsyncMapSequenceTests = TestSuite("AsyncMapSequence")
+
+AsyncMapSequenceTests.test("map values") {
+  let results = await [1, 2, 3].async.map { "\($0)" }.collect()
+  expectEqual(results, ["1", "2", "3"])
+}
+
+AsyncMapSequenceTests.test("map empty") {
+  let results = await [Int]().async.map { "\($0)" }.collect()
+  expectEqual(results, [])
+}
+
+AsyncMapSequenceTests.test("map throwing") {
+  let seq = [1, 2, 3].async
+    .map { value throws -> String in
+      if value == 2 {
+        throw Failure()
+      }
+      return "\(value)" 
+    }
+  var it = seq.makeAsyncIterator()
+  var results = [String]()
+  do {
+    while let value = try await it.next() {
+      results.append(value)
+    }
+    expectUnreachable()
+  } catch {
+    expectEqual(results, ["1"])
+  }
+}
+
+var AsyncFilterSequenceTests = TestSuite("AsyncFilterSequence")
```

Comment:
Need a throwing test for `filter`.

Reply:
the next couple of tests have em: I named the suites for both the non throwing and throwing variants.

e.g. the second and third tests both throw.

Reply:
I see the tests that call a `throws` closure, but I don't see any that actually throw, like the `"map throwing"` test, above.

Reply:
yep, got a whole glot of those posted as an update